macpkg/scripts/postinstall

```diff
@@ -1,15 +1,23 @@
 #!/usr/bin/env bash
 EXE_NAME="aws"
 COMPLETER_EXE_NAME="aws_completer"
-sudo mkdir -p /usr/local/bin
-# Files created here are not stored in the bill of materials for the pkg.
-# This means we cannot find and remove them during an installation. We add a
-# custom .install-metadata file here which is read and used by our
-# scripts/installers/uninstall-mac-pkg to clean up any extra files created
-# here. If any new files are added, be sure to add them to the .install-metadata
-# file as well.
-sudo echo "/usr/local/aws-cli/.install-metadata" >> /usr/local/aws-cli/.install-metadata
-sudo ln -sf "/usr/local/aws-cli/$EXE_NAME" "/usr/local/bin/$EXE_NAME"
-sudo echo "/usr/local/bin/$EXE_NAME" >> /usr/local/aws-cli/.install-metadata
-sudo ln -sf "/usr/local/aws-cli/$COMPLETER_EXE_NAME" "/usr/local/bin/$COMPLETER_EXE_NAME"
-sudo echo "/usr/local/bin/$COMPLETER_EXE_NAME" >> /usr/local/aws-cli/.install-metadata
+PATH_TO_INSTALL="$2/aws-cli"
+SYMLINK_DIR="/usr/local/bin"
+ID=$(id -u)
+
+# Only install symlinks if we are installing with root permissions
+# this means we selected the install for everyone option.
+if [ "$ID" == "0" ]; then
+    echo "Global install detected. Installing symlinks in $SYMLINK_DIR."
+    sudo mkdir -p $SYMLINK_DIR
+
+    AWS_PATH="$PATH_TO_INSTALL/$EXE_NAME"
+    AWS_LINK="$SYMLINK_DIR/$EXE_NAME"
+    echo "Linking $AWS_LINK -> $AWS_PATH "
+    sudo ln -sf $AWS_PATH $AWS_LINK
+
+    COMPLETER_PATH="$PATH_TO_INSTALL/$COMPLETER_EXE_NAME"
+    COMPLETER_LINK="$SYMLINK_DIR/$COMPLETER_EXE_NAME"
+    echo "Linking $COMPLETER_LINK -> $COMPLETER_PATH"
+    sudo ln -sf $COMPLETER_PATH $COMPLETER_LINK
```

Comment:
We should also start writing integration tests that actually run the installer in order to make sure that PKG functionality is not broken. Potential testing ideas would be:

* Test to make sure that this post-install script runs with sudo access.
* Be able to install it in a user location.